server/application/application.go

```diff
@@ -489,91 +486,98 @@ func (s *Server) ensurePodBelongsToApp(applicationName string, podName, namespac
 	if err != nil {
 		return nil, err
 	}
 	defer util.Close(closer)
-	return client.Resources(ctx, q)
+	return client.ResourcesTree(ctx, q)
 }
 
-func (s *Server) DeleteResource(ctx context.Context, q *ApplicationDeleteResourceRequest) (*ApplicationResponse, error) {
+func (s *Server) getAppResource(ctx context.Context, q *ApplicationResourceRequest) (*appv1.ResourceNode, *rest.Config, *appv1.Application, error) {
 	a, err := s.appclientset.ArgoprojV1alpha1().Applications(s.ns).Get(*q.Name, metav1.GetOptions{})
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
 	}
 	if !s.enf.Enforce(ctx.Value("claims"), rbacpolicy.ResourceApplications, rbacpolicy.ActionDelete, appRBACName(*a)) {
-		return nil, grpc.ErrPermissionDenied
+		return nil, nil, nil, grpc.ErrPermissionDenied
 	}
 
-	resources, err := s.getAppResources(ctx, &services.ResourcesQuery{ApplicationName: &a.Name})
+	resources, err := s.getAppResources(ctx, &services.ResourcesQuery{ApplicationName: a.Name})
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
 	}
 
 	found := findResource(resources.Items, q)
 	if found == nil {
-		return nil, status.Errorf(codes.InvalidArgument, "%s %s %s not found as part of application %s", q.Kind, q.APIVersion, q.ResourceName, *q.Name)
+		return nil, nil, nil, status.Errorf(codes.InvalidArgument, "%s %s %s not found as part of application %s", q.Kind, q.Group, q.ResourceName, *q.Name)
 	}
-	config, namespace, err := s.getApplicationClusterConfig(*q.Name)
+	config, _, err := s.getApplicationClusterConfig(*q.Name)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return found, config, a, nil
+}
+
+func (s *Server) GetResource(ctx context.Context, q *ApplicationResourceRequest) (*ApplicationResourceResponse, error) {
+	res, config, _, err := s.getAppResource(ctx, q)
 	if err != nil {
 		return nil, err
 	}
-	err = s.kubectl.DeleteResource(config, found, namespace)
+	obj, err := s.kubectl.GetResource(config, res.GroupKindVersion(), res.Name, res.Namespace)
 	if err != nil {
 		return nil, err
 	}
-	s.logEvent(a, ctx, argo.EventReasonResourceDeleted, fmt.Sprintf("deleted resource %s/%s '%s'", q.APIVersion, q.Kind, q.ResourceName))
-	return &ApplicationResponse{}, nil
+	data, err := json.Marshal(obj.Object)
+	if err != nil {
+		return nil, err
+	}
+	return &ApplicationResourceResponse{Manifest: string(data)}, nil
 }
 
-func (s *Server) Resources(ctx context.Context, q *services.ResourcesQuery) (*services.ResourcesResponse, error) {
-	a, err := s.appclientset.ArgoprojV1alpha1().Applications(s.ns).Get(*q.ApplicationName, metav1.GetOptions{})
+func (s *Server) DeleteResource(ctx context.Context, q *ApplicationResourceRequest) (*ApplicationResponse, error) {
+	res, config, a, err := s.getAppResource(ctx, q)
 	if err != nil {
 		return nil, err
 	}
+
 	if !s.enf.Enforce(ctx.Value("claims"), rbacpolicy.ResourceApplications, rbacpolicy.ActionGet, appRBACName(*a)) {
```

Comment:
The action is incorrect. The action is rbacpolicy.ActionGet but the method is deleting a resource

Reply:
Fixed